src/main.cpp

```diff
@@ -58,12 +58,18 @@ void intercept::on_frame() {
     for (auto& it : Threading::get().completedAsyncTasks) {
         __itt_task_begin(domainMain, __itt_null, __itt_null, main_on_frame_callback);
 
-        if (!it->data->callback.is_nil() && it->data->res) {
+        if (it->data->callbacks.size() != 0 && it->data->res) {
```

Comment:
```suggestion
        if (!it->data->callbacks.empty() && it->data->res) {
```
Atleast I think empty() is a method

Reply:
done